lib/samson/logging.rb

```diff
@@ -26,4 +26,26 @@
   require 'syslog/logger'
   config.logger = Syslog::Logger.new('samson')
   config.lograge.formatter = Lograge::Formatters::Logstash.new
+elsif ENV["SERVER_MODE"] # regular file logger that needs rotating
+  # Reopen logfile when we receive a SIGHUP for log-rotation
+  # puma normally kills itself with INT if it receives a SIGHUP
+  # see https://github.com/puma/puma/pull/1527 and https://github.com/puma/puma/blob/master/docs/signals.md
+  # We need to use self-pipe since we cannot reopen logs in an interrupt.
+  # This will break stdout_redirect feature of puma since we cannot support both
+  read, write = IO.pipe
+  Signal.trap(:SIGHUP) { write.puts }
+  Thread.new do
+    loop do
+      read.gets
+      Rails.logger.info "Received SIGHUP ... reopening logs"
+      # Rails.logger does not know what file it opened, so we help it https://github.com/rails/rails/issues/32211
+      dev = Rails.logger.instance_variable_get(:@logdev)&.dev
+      Rails.logger.reopen(dev.path)
```

Comment:
Isn't it possible for `dev` to be `nil` here, so you'd be calling `nil.path`?

Reply:
no since this must be a file logger at that point ... I'll remove the &. too that was ;eftover from previous approach